src/features/Apiexplorer/RequestJSONBox/__tests__/RequestJsonBox.test.tsx

```diff
@@ -80,22 +93,43 @@ describe('RequestResponseRenderer', () => {
       name: null as TSocketEndpointNames,
       auth: 0,
     };
-    cleanup();
     render(<RequestJSONBox {...newProps} />);
     const textarea = screen.getByLabelText('Request JSON');
     expect(textarea).toBeDisabled();
   });
 
+  it('should disable text box if no api call is selected in the dropdown', async () => {
+    const newProps = {
+      handleChange: jest.fn(),
+      request_example: 'asdfaewfaewfaewfd',
+      name: null as TSocketEndpointNames,
+      auth: 0,
+    };
+    render(<RequestJSONBox {...newProps} />);
+    const send = screen.getByRole('button', { name: /send request/i });
+    userEvent.click(send);
+
+    const popup = await screen.findByText(/your json object is invalid/i);
+    expect(popup).toBeVisible();
+
+    const close_icon = screen.getByAltText(/close-icon/i);
+    userEvent.click(close_icon);
+
+    await waitFor(() => expect(popup).not.toBeVisible());
+  });
+
   it('should render response renderer component', async () => {
+    render(<RequestJSONBox {...mockProps} />);
     const primaryButton = screen.getByRole('button', { name: /Send Request/i });
     const secondaryButton = screen.getByRole('button', { name: /clear/i });
-    await userEvent.click(primaryButton);
-    const playgroundSection = screen.getByTestId('dt_playground_section');
+    userEvent.click(primaryButton);
+    const playgroundSection = await screen.findByTestId('dt_playground_section');
```

Comment:
Do we need await here?

Reply:
Yes, after clicking the button, the dt_playground_section will appear. In order to find it with the test, we need to use await with `findby`. It's necessary to check this async.